src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -979,450 +960,188 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
 
 #End Region
 
-        ' at this point it is very likely that we are located at 
-        ' the beginning of a token        
-        Private Function TryScanToken(precedingTrivia As SyntaxList(Of VisualBasicSyntaxNode)) As SyntaxToken
-
-            If Not CanGet() Then
-                Return MakeEofToken(precedingTrivia)
-            End If
-
-            Dim ch As Char = Peek()
+        Private Function ScanTokenCommon(precedingTrivia As SyntaxList(Of VisualBasicSyntaxNode), ch As Char, fullWidth As Boolean) As SyntaxToken
+            Dim lengthWithMaybeEquals As Integer = 1
+            Dim c As Char
             Select Case ch
-                Case CARRIAGE_RETURN, LINE_FEED, NEXT_LINE, LINE_SEPARATOR, PARAGRAPH_SEPARATOR
+                Case CARRIAGE_RETURN, LINE_FEED
                     Return ScanNewlineAsStatementTerminator(ch, precedingTrivia)
-
+                Case NEXT_LINE, LINE_SEPARATOR, PARAGRAPH_SEPARATOR
+                    If Not fullWidth Then
+                        Return ScanNewlineAsStatementTerminator(ch, precedingTrivia)
+                    End If
                 Case " "c, CHARACTER_TABULATION, "'"c
                     Debug.Assert(False, String.Format("Unexpected char: &H{0:x}", AscW(ch)))
                     Return Nothing ' trivia cannot start a token
-
                 Case "@"c
-                    Return MakeAtToken(precedingTrivia, False)
-
+                    Return MakeAtToken(precedingTrivia, fullWidth)
```

Comment:
I think the goal for hardcoded "false" was the expectation that JIT would inline MakeAtToken and constant-propagate the "false", so only the codepath for "false" remains. Nonfullwidth scenarios are by far the most common case and the goal was to make it as fast as possible. Need to verify if that actually the case, since in such case the unified implementation could be slower.
Some scenarios in IDE can get fairly sensitive to the throughput of parsing/scanning, so any extra perf, even at the cost of maintainability (within reason) is considered a good tradeoff here, so must be careful that we are not doing the opposite.